### ISSUE TITLE
[Frontend] Add "save" component

### ISSUE THREAD

**Issue by joofsh** [department-of-veterans-affairs/caseflow#2475]
Create component for save loading spinner & save callback mechanism

#### Acceptance Criteria
- Create component that shows the saving text & loading spinner
- Save event should be triggered on any inputs:
  - before page close 
  - Every X seconds, trigger a function that determines if a save is necessary. If so, call the save action &  display the saving text. For Hearing Prep this configurable property should be set to 30 seconds
- Under these conditions, the component should trigger a callback function. This function should be stubbed out for now and only console log that the callback was triggered

Note:
- Test using the Daily Docket page
- Don't make any backend calls. You can stub out an API call.

**Comment by slj** [comment]
After some thought, I've prototyped a simpler component than the one requested.  It cares about 3 properties only:
- `app` (string name of the app / used to determine spinner color)
- `isSaving` (boolean)
- `beforeWindowClosesActionCreator` (function)

The idea is that this component doesn't know anything about timings or other UI events.  If you call it with `true` it renders "Saving..." with an app-specific-colored spinner.  If `false`, it renders "Last saved at <time>".  If you close the window, it dispatches the action passed to it.  When to save (on blurs/after X ms/sec) is a concern more easily handled outside this component IMO.

Furthermore, I think my version is much easier to test in karma.

I already have this working in a locally:
![save-component](https://user-images.githubusercontent.com/1053255/29425227-82f6d0b2-8350-11e7-84d6-5b4fb84dc414.gif)


But to show you this, I had to modify elements outside the component, thus going beyond the scope of this ticket.   I'd like to trim the fat and submit a PR based on the AC I've described.  May I proceed?
